### PR TITLE
tetragon: resolve uid to username for exec events from /proc fs

### PR DESCRIPTION
**Pull request by tixxdz**
```release-note
tetragon: resolve uid to username for exec events from /proc fs
```

### CODE REVIEW

File: pkg/sensors/exec/userinfo/userinfo.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
```
Comment (kkourt) [inline pkg/sensors/exec/userinfo/userinfo.go:2]: The first commit is named  "tetragon: resolve uid->username for processes from /proc", but it seems that it just refactors the code for subsequent patches. Please rename the commit. Also, all of the patches have a single tile. Please add some context what each patch does and why.

File: pkg/sensors/exec/userinfo/userinfo.go [hunk 2]
```diff
@@ -37,6 +38,7 @@ func MsgToExecveAccountUnix(m *exec.MsgExecveEventUnix) error {
 	if option.Config.UsernameMetadata == int(option.USERNAME_METADATA_UNIX) {
 		username, err := getAccountUnix(m.Unix.Process.UID, &m.Unix.Msg.Namespaces)
 		if err != nil {
+			errormetrics.ErrorTotalInc(errormetrics.ProcessMetadataUsernameFailed)
```
Comment (kkourt) [inline pkg/sensors/exec/userinfo/userinfo.go:41]: Shouldn't we check here if the error is ErrNotInHostNs?

**Review by kkourt — CHANGES_REQUESTED** [review]
Thanks, please find some comments below.

File: pkg/sensors/exec/userinfo/userinfo.go (hunk 1 above)
Comment (tixxdz) [reply pkg/sensors/exec/userinfo/userinfo.go:2]: Actually it is used in this patch, but improved git log.

File: pkg/sensors/exec/userinfo/userinfo.go (hunk 2 above)
Comment (tixxdz) [reply pkg/sensors/exec/userinfo/userinfo.go]: Makes sense, I made it 'process_metadata_username_ignored_not_in_host_namespaces' a bit too long but it has to be descriptive anyway.

**Comment by tixxdz** [comment]
> Thanks, please find some comments below.

Done, thank you, PTAL!

**Comment by tixxdz** [comment]
backport is here: https://github.com/cilium/tetragon/pull/2603 , when merged will

**Review by kkourt — APPROVED** [review]

**Pull request merged**